go/pkg/client/cas_upload.go

```diff
@@ -246,9 +246,28 @@ func (c *Client) uploadUnified(ctx context.Context, entries ...*uploadinfo.Entry
 		return nil, 0, err
 	}
 	wait := make(chan *uploadResponse, uploads)
-	var missing []digest.Digest
+	var dgs []digest.Digest
+	dedupDgs := make(map[digest.Digest]bool, uploads)
+	for _, ue := range entries {
+		if _, ok := dedupDgs[ue.Digest]; !ok {
+			dgs = append(dgs, ue.Digest)
+			dedupDgs[ue.Digest] = true
+		}
+	}
```

Comment:
Is this deduplication necessary? It used to be so when it happened after aggregating digests from multiple calls, but since this is now here, it doesn't seem likely to be duplicates from a single caller. Even if there was, checking for non-missing within the upload loop below renders this loop even less useful.